lib/prompts/createProjectPrompt.ts

```diff
@@ -117,10 +140,19 @@ export async function createProjectPrompt(
 
   if (providedTemplateIsValid) {
     result.projectTemplate = findTemplateByNameOrLabel(
-      projectTemplates,
+      projectTemplates!,
       promptOptions.template!
     );
   }
 
-  return result;
-}
+  if (projectTemplates && projectTemplates.length > 0) {
+    if (!result.projectTemplate) {
+      throw new Error(
+        'Project template is required when projectTemplates is provided'
+      );
+    }
+    return result as CreateProjectPromptResponseWithTemplate;
+  }
+
+  return result as CreateProjectPromptResponseWithoutTemplate;
+} as CreateProjectPromptOverload;
```

Comment:
Are we actually getting anything out of these new types with all this casting happening? Is it possible to do this without casting?

Reply:
Thanks for calling this one out. I had initially tried a bunch of different variations to get this working, but I just took another look and I can do it without any casting at all. I fixed it in [85c3a79](https://github.com/HubSpot/hubspot-cli/pull/1449/commits/85c3a79987a57b1031cb52e71a1eb499a5612805). Turns out I'm just not very familiar with how overloads work 😅